src/Xamarin.Android.Build.Tasks/Properties/Resources.resx

```diff
@@ -137,6 +137,10 @@
     <value>Ignoring Reference Assembly `{0}`.</value>
     <comment>{0} - File name of the assembly.</comment>
   </data>
+  <data name="XA0122" xml:space="preserve">
+    <value>Assembly '{0}' is using a deprecated attribute '[assembly: {1}]'. Use a newer version of this NuGet package or notify the library author.</value>
```

Comment:
Instead of a generic "XA0122" warning for "deprecated attribute(s)", this should instead be a warning specific to `Java.Interop.DoNotPackageAttribute` *so that* the warning message can specifically suggest the use of the `@(AndroidExternalLibrary)` Build action as a replacement.

Reply:
Peppers might have some good ideas on ways to cover both audiences, but one little tricky thing about this warning is that it can appear either for library authors or library consumers. For the library consumers, the information about `@(AndroidExternalLibrary)` would potentially be confusing.

Based on the GitHub search results, I'm cautiously optimistic there will be almost no one who will see this warning. I searched around a little to gather a little more info on the Xamarin authored NuGet packages that used the attribute too. From what I could tell, they are delisted now. (An example is Xamarin.Amazon.DeviceMessaging.)

Reply:
I think I will make `[Obsolete]` mention `@(AndroidExternalLibrary)` so NuGet authors get that message.

The other message seems appropriate, I think.